### PR TITLE
Fix inconsistency in keeping track of NuGetVersion.OriginalVersion

### PR DESCRIPTION
**Pull request by xavierdecoster**
Fixes NuGet/Home#5068

### CODE REVIEW

File: src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public NuGetVersion(string version)
         /// Creates a NuGetVersion from an existing NuGetVersion
         /// </summary>
         public NuGetVersion(NuGetVersion version)
-            : this(version.Version, version.ReleaseLabels, version.Metadata, version.ToString())
+            : this(version.Version, version.ReleaseLabels, version.Metadata, version.OriginalVersion)
```
Comment (xavierdecoster) [inline src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs:31]: This was the culprit.

File: test/NuGet.Core.Tests/NuGet.Versioning.Test/NuGetVersionTest.cs [hunk 2]
```diff
@@ -289,6 +289,20 @@ public void ToStringConstructedFromVersionAndSpecialVersionConstructor(string ve
         }
 
         [Theory]
+        [InlineData("01.42.0")]
+        [InlineData("01.0")]
+        [InlineData("01.42.0-alpha")]
+        [InlineData("01.42.0-alpha.1")]
```
Comment (xavierdecoster) [inline test/NuGet.Core.Tests/NuGet.Versioning.Test/NuGetVersionTest.cs:295]: Without the above fix, these three SemVer2 versions would fail the test.

**Review by joelverhagen — APPROVED** [review]
Please wait for a client person's sign-off (@emgarten).

**Review by emgarten — APPROVED** [review]
LGTM

**Pull request merged**